### ISSUE TITLE
[icetray] implement __nonzero__ for POD types (Trac #1576)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1568]
`__nonzero__()` is used to test True/False in python. Let's implement that, so we can do this:

```text
if not I3Bool(False):
  print 'bool is false'
if not I3Int(0):
  print 'int is false'
```

Migrated from https://code.icecube.wisc.edu/ticket/1576
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:42", 
    "description": "`__nonzero__()` is used to test True/False in python. Let's implement that, so we can do this:\n\n{{{\nif not I3Bool(False):\n  print 'bool is false'\nif not I3Int(0):\n  print 'int is false'\n}}}\n", 
    "reporter": "david.schultz", 
    "cc": "olivas, nega", 
    "resolution": "fixed", 
    "_ts": "1550067102580394", 
    "component": "combo core", 
    "summary": "[icetray] implement __nonzero__ for POD types", 
    "priority": "major", 
    "keywords": "", 
    "time": "2016-03-04T19:59:05", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160314T19:17:54: 
* david.schultz changed status from "new" to "closed"
* david.schultz commented:

> In changeset 143162:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="143162"
> implement __nonzero__ for I3Double. Along with r2464/Icetray, closes #1568
> ```


* david.schultz changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:42: 
* olivas changed milestone from "April 2016" to ""
* olivas commented:

> Milestone April 2016 deleted
